cli/src/diff_util.rs

```diff
@@ -830,19 +844,29 @@ fn materialized_diff_stream<'a>(
         .buffered((store.concurrency() / 2).max(1))
 }
 
+fn map_access_denied(err: BackendError) -> CommandError {
+    match err {
+        BackendError::ReadAccessDenied { .. } => user_error_with_hint(
+            err,
+            "The `--color-words` format supports access-restricted paths.",
+        ),
+        other => other.into(),
+    }
+}
```

Comment:
Can you move this to command_error.rs?

I've added `DiffRenderError` because `CommandError` can't be easily handled in templater.